src/main/CMakeLists.txt

```diff
@@ -78,9 +78,11 @@ endif()
 #   https://cmake.org/cmake/help/v3.0/prop_tgt/LINK_SEARCH_START_STATIC.html
 #   https://cmake.org/cmake/help/v3.0/prop_tgt/LINK_SEARCH_END_STATIC.html
 if(ENABLE_STATIC_BINARY)
-  set_target_properties(cvc5-bin PROPERTIES LINK_FLAGS -static)
-  set_target_properties(cvc5-bin PROPERTIES LINK_SEARCH_START_STATIC ON)
-  set_target_properties(cvc5-bin PROPERTIES LINK_SEARCH_END_STATIC ON)
+  if(NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    set_target_properties(cvc5-bin PROPERTIES LINK_FLAGS -static)
+    set_target_properties(cvc5-bin PROPERTIES LINK_SEARCH_START_STATIC ON)
+    set_target_properties(cvc5-bin PROPERTIES LINK_SEARCH_END_STATIC ON)
+  endif()
```

Comment:
Our current solution for this problem is that on macOS, we always compile with `--static --no-static-binary` if we want almost-static binaries. I think that that is the better solution because it does not silently change what the user asked for.